lib/math.ex

```diff
@@ -563,4 +563,123 @@ defmodule Math do
       _ -> raise ArithmeticError, "Inputs are not coprime!"
     end
   end
+
+  # Interpolation
+
+  @doc """
+  Computes the y value of a given t point on a bezier_curve
+
+      ## Examples
+
+      iex> Math.bezier_curve(0.5, [{0,0}, {1,1}])
+      {0.5,0.5}
+
+  """
+  def bezier_curve(t, control_points) when is_list(control_points)do
+
+    new_points =
+      control_points
+    |> Enum.with_index()
+    |> Enum.slice(0..-2)
+    |> Enum.map(
+      fn {p0, index} ->
+          next_index = index + 1
+          p1 = Enum.at(control_points, next_index)
+          linear_interpolation(t, p0, p1)
+      end
+    )
+
+    if Enum.count(new_points) == 1 do
+      Enum.at(new_points, 0)
+    else
+      bezier_curve(t, new_points)
+    end
+  end
+
+  @doc """
+  Computes the y value of a given t point on a bezier_curve
+
+  Similar to `bezier_curve/2`, but raises an error if not on range [0,1].
+
+    ## Examples
+
+      iex> Math.bezier_curve!(0.5, [{0,0}, {1,1}])
+      {0.5,0.5}
+
+      iex> Math.bezier_curve!(1.5, [{0,0}, {1,1}])
+      ** (ArgumentError) t is not beetween 0 and 1
+  """
+
+  def bezier_curve!(t, control_points)
+    when is_number(t)
+    and 0.0 <= t
+    and t <= 1.0
+    and is_list(control_points) do
+      bezier_curve(t, control_points)
+  end
+
+  def bezier_curve!(t, control_points)
+    when is_number(t)
+    and is_list(control_points) do
+      raise ArgumentError, "t is not beetween 0 and 1"
+  end
+
+  @doc """
+  Computes the y value of a given t point on a linear_interpolation between 2 points
+
```

Comment:
Explain what happens when points are outside of the range `0..1` (i.e. the difference between this and the bang-version).